recipe-generator/src/main/java/ca/derekcormier/recipe/generator/TypescriptFilters.java

```diff
@@ -0,0 +1,120 @@
+package ca.derekcormier.recipe.generator;
+
+import java.util.Map;
+
+import ca.derekcormier.recipe.cookbook.Cookbook;
+import ca.derekcormier.recipe.cookbook.CookbookUtils;
+import ca.derekcormier.recipe.cookbook.Primitive;
+import ca.derekcormier.recipe.cookbook.type.ArrayType;
+import ca.derekcormier.recipe.cookbook.type.EnumType;
+import ca.derekcormier.recipe.cookbook.type.FlagType;
+import ca.derekcormier.recipe.cookbook.type.ParamType;
+import ca.derekcormier.recipe.cookbook.type.PrimitiveType;
+import ca.derekcormier.recipe.cookbook.type.Type;
+import liqp.filters.Filter;
+
+public class TypescriptFilters {
+    public static Filter createTsTypeFilter(Cookbook cookbook) {
+        return new Filter("tstype") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                ParamType type = CookbookUtils.parseType(super.asString(value), cookbook);
+                return TypescriptFilters.toTsType(type);
+            }
+        };
+    }
+
+    public static Filter createTsParamFilter(Cookbook cookbook) {
+        return new Filter("tsparam") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                Map<String,String> param = (Map<String,String>)value;
+                String name = param.get("name");
+                ParamType type = CookbookUtils.parseType(param.get("type"), cookbook);
+
+                return (type.isVararg() ? "..." : "") + name + ": " + TypescriptFilters.toTsType(type);
+            }
+        };
+    }
+
+    public static Filter createJsParamFilter(Cookbook cookbook) {
+        return new Filter("jsparam") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                Map<String,String> param = (Map<String,String>)value;
+                String name = param.get("name");
+                ParamType type = CookbookUtils.parseType(param.get("type"), cookbook);
+
+                return (type.isVararg() ? "..." : "") + name;
```

Comment:
is this JS (not TS)? JS has no `...`/varargs...

Reply:
Actually I'm wrong, it is an es6 thing... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters